### PR TITLE
[feat] Add NetworkManager:isConnected()

### PR DESCRIPTION
**Pull request by Frenzie**
This enables connecting to calibre and FTP without Internet connectivity.

Pinging @robert00s @mk-f

### CODE REVIEW

File: frontend/ui/network/manager.lua [hunk 1]
```diff
@@ -60,6 +60,15 @@ function NetworkMgr:promptWifiOff(complete_callback)
     })
 end
 
+function NetworkMgr:isConnected()
+    if Device:isAndroid() then
+        return self:isWifiOn()
+    else
+        -- `-c1` try only once; `-w2` wait 2 seconds
+        return os.execute("ping -c1 -w2 $(ip route | grep '^default' | awk '/eth0/ {print $3}')") == 0
```
Comment (robert00s) [inline frontend/ui/network/manager.lua]: I'm not sure if `eth0` is a good idea. On my laptop, the network card is `wls3`. Maybe something like that?
`return os.execute("ping -c1 -w2 $(ip route | grep '^default' | awk '/default/ {print $3}')") == 0`

Comment (Frenzie) [reply frontend/ui/network/manager.lua]: The question is more what it is on your Kindle than what it is on your laptop. :-P

But yes, my original proposal was `ping -c1 -w2 $(ip route | awk '/default/ { print $3 }')` which should be fine.

Comment (robert00s) [reply frontend/ui/network/manager.lua]: On my Kindle is `wlan0` :)

**Comment by mk-f** [comment]
Uh, this is a problem on PB, because:
```sh
~ # ip
-sh: ip: not found
~ # find / -name ip
~ # 
```
An alternative could be:
```sh
 /sbin/route -n | awk '$4 == "UG" {print $2}'
```

**Comment by Frenzie** [comment]
That would work on Kobo. Same for Kindle @robert00s ?

**Comment by robert00s** [comment]
Looks good.
Should work :) I don't have local FTP server so I don't test.

**Comment by Frenzie** [comment]
Alright, going ahead and merging this.

**Pull request merged**